modules/nf-core/bakta/bakta/main.nf

```diff
@@ -35,6 +35,8 @@ process BAKTA_BAKTA {
     def proteins_opt = proteins ? "--proteins ${proteins[0]}" : ""
     def prodigal_tf = prodigal_tf ? "--prodigal-tf ${prodigal_tf[0]}" : ""
     """
+    amrfinder_update --force_update --database $db/amrfinderplus-db
```

Comment:
Is this required for BATKA?

Reply:
Tests now pass without this line. I had to add it because some DB files are not in the downloaded folder but are created by `amrfinder_update` (similar to the antismash vs. antismash-lite issue). I don't know what changed but I'll remove the line. If I ever encounter DB issues with bakta I hopefully remember this comment/commit ;)

Reply:
:tada:

Reply:
Hi @jfy133 and @jasmezz,

I am running into this issue with the `amrfinder_update`. The `BAKTA_BAKTADBDOWNLOAD` command fails to download the amrfinderplus-db. So when `bakta` tries to find this database, it fails:

```nextflow
  Traceback (most recent call last):
    File "/local/cluster/micromamba/envs/bakta-1.9.1/bin/bakta", line 10, in <module>
      sys.exit(main())
    File "/local/cluster/micromamba/envs/bakta-1.9.1/lib/python3.8/site-packages/bakta/main.py", line 275, in main
      expert_amr_found = exp_amr.search(cdss, cds_aa_path)
    File "/local/cluster/micromamba/envs/bakta-1.9.1/lib/python3.8/site-packages/bakta/expert/amrfinder.py", line 47, in search
      raise Exception(f"amrfinder error! error code: {proc.returncode}. Please, try 'amrfinder_update --force_update --database {amrfinderplus_db_path}' to update AMRFinderPlus's internal database.")
  Exception: amrfinder error! error code: 1. Please, try 'amrfinder_update --force_update --database /nfs5/BPP/Grunwald_Lab/home/paradarc/pathogensurveillance/work/3d/19178e85f70539076ec38ddcf5bd0f/db/amrfinderplus-db' to update AMRFinderPlus's internal database.
``` 
Do you recommend to add this line to the backta's `main.nf`? or do you have any idea on how to get it to work?

` amrfinder_update --force_update --database $db/amrfinderplus-db`

For context, I working on the [pathogensurveillance](https://github.com/nf-core/pathogensurveillance) pipeline.

Thanks!

Camilo

Reply:
Hi @cahuparo, did you use bakta (and baktadbdownload) with default parameters or change anything? I am trying to replicate your error, currently unsuccessfully (using either conda or singularity for the test runs). I would like to have a reproducible case before adding the `amrfinder_update` line back in, because that would slow the module down.

Reply:
Hi @jasmezz, 

when I ran the pipeline from a brand new directory I saw the error I showed above. For that, my command was:

```bash
nextflow run main.nf --input https://raw.githubusercontent.com/grunwaldlab/pathogensurveillance/master/test/data/metadata_small.csv --outdir test_out --download_bakta_db true -profile mamba -resume
``` 
So I added the line for the `amrfinder_update` to the bakta `main.nf` this way:

```nextflow
    """
    /nfs5/BPP/Grunwald_Lab/home/paradarc/pathogensurveillance/work/conda/bakta_baktadbdownload-cfec0aac1e1132b000975a91ee5beb7d/bin/amrfinder_update --force_update --database $db/amrfinderplus-db

    bakta \\
        $fasta \\
        $args \\
        --threads $task.cpus \\
        --prefix $prefix \\
        $proteins_opt \\
        $prodigal_tf \\
        --db $db

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bakta: \$(echo \$(bakta --version) 2>&1 | cut -f '2' -d ' ')
    END_VERSIONS
    """
``` 
Adding the line actually worked. So we are debating if we need to create a local `main.nf` to handle that. But seems to only happen with the conda/mamba installation for some reason.

Thanks, 

Camilo

Reply:
Hi @cahuparo, I finally found time for this! I fixed it in #5048.